ronn/geopm_agent_frequency_map.7.ronn

```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
-
-  To encode as a json string, include a series of HASH_{index}, FREQ_{index}
-  pairs in the policy. Since json does not support hex literals, input the
-  hashes in their decimal form. For example, the above policy would look like:
-  { FREQ_MIN: 1e9, FREQ_MAX: 3e9, HASH_0: 305441741, FREQ_0: 2e9,
-    HASH_1: 610848239, FREQ_1: 2.5e9 }
-
-  The index values can be in the range of 0 to 30, inclusive. The order of
-  index values does not matter, and gaps in index values are permitted. Multiple
-  definitions of an index are not permitted, and multiple mappings of a region
-  are not permitted.
-
-  If the GEOPM_FREQUENCY_MAP environment variable specifies a frequency map,
-  the policy only consists of FREQ_MIN and FREQ_MAX. GEOPM_FREQUENCY_MAP is
-  deprecated and will no longer be used in a future release.
-
-
-* `Agent Sample Definitions`:
-  N/A
-
-* `Trace Column Extensions`:
-  N/A
-
-* `Report Extensions`:
-  The assigned frequency for each region is added to the report.
-
-* `Control Loop Gate`:
-  The agent gates the Controller's control loop to a cadence
-  of 5ms.
+The frequency map agent selects CPU frequency in response to changes
+in the application region being executed on each CPU.  The agent
+policy provides a map from a GEOPM region hash to a CPU frequency, see
+**geopm_hash(3)** for more information about the GEOPM region name
+hash function.  When a CPU is executing a region with a hash provided
+by the policy, the CPU will operate at the mapped frequency.  The
+policy also provides a default CPU frequency which is applied to all
+CPUs running the application while not executing a region described by
+the policy map.
+
+Applying any control algorithm that changes CPU frequency on short
+intervals can result in undesirable interactions with the hardware
+control algorithm that dynamically selects uncore frequency.  For this
+reason, the frequency map agent policy provides an option to fix the
+uncore frequency to a single value while executing the application.
+If this option is not specified, the uncore frequency is not
+controlled by the agent at all.
+
+Note: this agent can be used to control the application CPUs and
+optionally the uncore to run at fixed frequencies if no region
+frequency map is provided.
+
+
+## AGENT NAME
+The agent described in this manual is selected in many geopm
+interfaces with the "frequency_map" agent name.  This name can be
+passed to **geopmlaunch(1)** as the argument to the `--geopm-agent`
+option, or the `GEOPM_AGENT` environment variable can be set to this
+name (see **geopm(7)**).  This name can also be passed to the
+**geopmagent(1)** as the argument to the '-a' option.
+
+
+## POLICY PARAMETERS
+
+  `FREQ_DEFAULT`: The operating frequency set by the agent on CPUs
+                  executing the application while the application is
+                  not within a policy-mapped region.  The processor's
+                  sticker frequency is selected for this parameter if
```

Comment:
this means that running with this agent with NAN policy only could have lower performance than the monitor.